homework/index.js

```diff
@@ -30,14 +30,104 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const listContainer = document.getElementById('repo_information_table-container');
+    clearContainer(listContainer);
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert alert-error' });
+  }
+
+  function addRow(tbody, label, value = '') {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { text: `${label} :`, class: 'label' });
+    createAndAppend('td', row, { text: value });
+    return row;
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    addRow(tbody, 'Description:', repository.description);
+    addRow(tbody, 'Forks:', repository.forks);
+    addRow(tbody, 'Updated:', new Date(repository.updated_at).toLocaleString());
+  }
+
+  function createContributorsList(repository, ul) {
+    clearContainer(ul);
+    const contributorsUrl = repository.contributors_url;
+    fetchJSON(contributorsUrl, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const a = createAndAppend('a', li, {
+          href: repository.html_url,
+          target: '_blank',
+          id: 'link_contributions',
+        });
+        const div = createAndAppend('div', a);
+
+        createAndAppend('img', div, {
+          class: 'contributor_avatar',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        createAndAppend('p', div, { id: 'p1', text: contributor.login });
```

Comment:
I saw a new issue: because you are creating this in a loop you end up with multiple occurrences of  **id** `p1` and `p2`. In HTML ids must be unique (occur once only). So in this case, you need to use CSS classes instead of ids.

Reply:
applied